protocol/auditlog.go

```diff
@@ -11,10 +11,21 @@ import (
 )
 
 type directoryHistory struct {
+	*AudState
 	addr      string
-	signKey   sign.PublicKey
-	snapshots map[uint64]*DirSTR
-	latestSTR *DirSTR
+	snapshots []*DirSTR
```

Comment:
Looking at current code, there are insertion and search operations, which are O(1), so I think it wouldn't be a huge performance issue.